Removing legacy pub_ddr init code

jbech-linaro opened the pull request:
Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org>
Suggested-by: Sandeep Tripathy <sandeep.tripathy@broadcom.com>

jbech-linaro commented:
This resolves #806 , @sandeepbrcm I've added your name as _suggested by_, let me know if that is OK or not.

sandeepbrcm commented:
Thanks. perfectly fine.

jenswi-linaro commented on core/arch/arm/mm/tee_mmu.c:
```diff
@@ -668,18 +667,6 @@ void teecore_init_pub_ram(void)
-	 */
-	TEE_ASSERT((e - s) > nsec_tee_size);
-
-	TEE_ASSERT(tee_mm_is_empty(&tee_mm_pub_ddr));
```
I suppose it should be removed from `core/arch/arm/mm/tee_mm.c` too.

jbech-linaro commented:
I've removed the rest of the old stuff related to the secure side usage. At the same time I updated naming(s). Please let me know if you prefer having it split into two patches instead of just one.

jenswi-linaro commented on core/arch/arm/include/mm/core_mmu.h:
```diff
@@ -122,9 +122,12 @@ enum teecore_memtypes {
 };
 
-/* Default NSec shared memory allocated from NSec world */
-extern unsigned long default_nsec_shm_paddr;
```
This old name is more clear than the new, especially since nsec_shm or NSEC_SHM is used at many places.

jbech-linaro replied:
Maybe, my intention here was to make it similar to the tee_mm's we have in tee_mm.c. When we are dealing with ddr on secure side we're using `tee_mm_sec_ddr`. So I thought having the same for public (but without the "_mm_", since it's not handle in there directly) would be more appropriate.

The thing is that we have so many ways in the code to talk about memory and I think that is very confusing. I would like to go over all those functions, variables, enums, cfg's etc to see if you can make them more common. A quick grep and sort and pick a few names (there's plenty more) containing ddr, nsec, sec, ram, mem gives:
```
tee_mm_pub_ddr
tee_mm_sec_ddr
CFG_DDR_START
CFG_DDR_TEETZ_RESERVED_START
CFG_DDR_TEETZ_RESERVED_SIZE
DDR_PHYS_START
DDR_SIZE
TA_FLAG_EXEC_DDR
CFG_TA_RAM_SIZE
TZDRAM_BASE
CFG_DDR_SIZE
CFG_DDR_ARMTZ_ONLY_START
CFG_DDR_ARMTZ_ONLY_SIZE
CFG_TEE_RAM_START
sec_ddr_attr
teecore_init_pub_ram
CORE_MEM_NSEC_SHM
MEM_AREA_NSEC_SHM
MEM_AREA_TEE_RAM,                                                                                                                                                        
MEM_AREA_TEE_COHERENT,                                                                                                                                                   
MEM_AREA_TA_RAM,                                                                                                                                                         
MEM_AREA_NSEC_SHM,                                                                                                                                                       
MEM_AREA_KEYVAULT,                                                                                                                                                       
MEM_AREA_IO_SEC,                                                                                                                                                         
MEM_AREA_IO_NSEC,                                                                                                                                                        
MEM_AREA_TA_VASPACE
map_nsec_shm
```
I know that doing such a comparison is a little like mixing apples and oranges. But they are related in some sense and it would be nice to have a common naming schema/convention.

Having that said, I could skip the renaming in this patch and then attack it later on, in a separate refactoring patch if that is preferred?

jenswi-linaro replied:
Yes please.

jenswi-linaro commented:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

jforissier commented:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

I'm addressing the Travis build error in a separate PR (broken server `releases.linaro.org`).

jbech-linaro commented:
Rebased, tags applied (and I took the liberty of adding a bit more text to the commit message).

Regarding (Travis) build errors I already have something there:
https://github.com/OP-TEE/build/pull/66
https://travis-ci.org/jbech-linaro/optee_os/builds/137455243 (soon a PR if it works)

jbech-linaro commented:
Rebased it and (force)pushed it again. Hopefully Travis should become "green" now, since Jerome's toolchain fix has been merged.

jbech-linaro commented:
And a new patch pushed, this time with checkpatch fixes that Travis revealed.
